README.md
# Black-Webbrowser `v1.5`

[![image](https://user-images.githubusercontent.com/78996423/139568252-acba2954-a5be-4a5f-8b46-790b35e45ed9.png)](https://github.com/black-software-Com/Black-Webbrowser)


``` txt

 ▄▄▄▄    ██▓    ▄▄▄       ▄████▄   ██ ▄█▀    █     █░▓█████  ▄▄▄▄    ▄▄▄▄    ██▀███   ▒█████   █     █░  ██████ ▓█████  ██▀███  
▓█████▄ ▓██▒   ▒████▄    ▒██▀ ▀█   ██▄█▒    ▓█░ █ ░█░▓█   ▀ ▓█████▄ ▓█████▄ ▓██ ▒ ██▒▒██▒  ██▒▓█░ █ ░█░▒██    ▒ ▓█   ▀ ▓██ ▒ ██▒
▒██▒ ▄██▒██░   ▒██  ▀█▄  ▒▓█    ▄ ▓███▄░    ▒█░ █ ░█ ▒███   ▒██▒ ▄██▒██▒ ▄██▓██ ░▄█ ▒▒██░  ██▒▒█░ █ ░█ ░ ▓██▄   ▒███   ▓██ ░▄█ ▒
▒██░█▀  ▒██░   ░██▄▄▄▄██ ▒▓▓▄ ▄██▒▓██ █▄    ░█░ █ ░█ ▒▓█  ▄ ▒██░█▀  ▒██░█▀  ▒██▀▀█▄  ▒██   ██░░█░ █ ░█   ▒   ██▒▒▓█  ▄ ▒██▀▀█▄  
░▓█  ▀█▓░██████▒▓█   ▓██▒▒ ▓███▀ ░▒██▒ █▄   ░░██▒██▓ ░▒████▒░▓█  ▀█▓░▓█  ▀█▓░██▓ ▒██▒░ ████▓▒░░░██▒██▓ ▒██████▒▒░▒████▒░██▓ ▒██▒
░▒▓███▀▒░ ▒░▓  ░▒▒   ▓▒█░░ ░▒ ▒  ░▒ ▒▒ ▓▒   ░ ▓░▒ ▒  ░░ ▒░ ░░▒▓███▀▒░▒▓███▀▒░ ▒▓ ░▒▓░░ ▒░▒░▒░ ░ ▓░▒ ▒  ▒ ▒▓▒ ▒ ░░░ ▒░ ░░ ▒▓ ░▒▓░
▒░▒   ░ ░ ░ ▒  ░ ▒   ▒▒ ░  ░  ▒   ░ ░▒ ▒░     ▒ ░ ░   ░ ░  ░▒░▒   ░ ▒░▒   ░   ░▒ ░ ▒░  ░ ▒ ▒░   ▒ ░ ░  ░ ░▒  ░ ░ ░ ░  ░  ░▒ ░ ▒░
 ░    ░   ░ ░    ░   ▒   ░        ░ ░░ ░      ░   ░     ░    ░    ░  ░    ░   ░░   ░ ░ ░ ░ ▒    ░   ░  ░  ░  ░     ░     ░░   ░ 
 ░          ░  ░     ░  ░░ ░      ░  ░          ░       ░  ░ ░       ░         ░         ░ ░      ░          ░     ░  ░   ░     
      ░                  ░                                        ░       ░                                              
```
<hr>

File | Windows | Linux | MacOS |
:--------:|:-------:|:-----:|:------:|
[Black-Webbrowser](https://github.com/black-software-com/black-webbrowser) v1.5| ✅ | ✅ | ✅ |
[Black-Webbrowser](https://github.com/mrprogrammer2938/black-webbrowser) v1.0 | ✅ | ✅ | ✅ |
<br>
<center>
<a href='https://github.com/black-software-Com/Black-Webbrowser/archive/refs/heads/master.zip' target="_top" title='Download'>
 <img src='./Scr/download-logo.png' width=100>
 </a></center>
<br>

**Installing**
``` sh
$ git clone https://github.com/black-software-com/Black-Webbrowser

$ cd Black-Webbrowser

$ bash install.sh
```
<br>

*Argument*
``` txt
Black-Webbrowser Argument:
        python black 
        or
        python black <URL>
        
        python black --version | version
        
        python black --help | help
```
<br>

**Update**
``` sh
$ cd /Update

$ python update.py

```

**Uninstalling**
``` sh
$ cd Uninstall

$ python uninstall.py
```

#### [Instagram](https://instagram.com/black_software_company)

#### [Black](https://github.com/mrprogrammer2938/Black-Webbrowser)
<br>

##### [License](https://github.com/black-software-Com/Black-Webbrowser/blob/master/LICENSE)

#### If the software has a problem, Email the address below
``` txt 
blacksoftwarecompany3@gmail.com
```

#### or

#### [issues](https://github.com/black-software-Com/Black-Webbrowser/issues)
<br>


history.txt

github.com
github.com
github.com


Core/install.py
#!/usr/bin/python3
# Black-Webbrowser v1.5
import os,subprocess
try:
    from tkinter import *
    from tkinter.ttk import *
    from ttkbootstrap import Style
except ImportError:
    subprocess.getoutput("pip install tk-tools")

class black_webbrowser(Tk):
    def __init__(self):
        super(black_webbrowser,self).__init__()
        self.title('Black-Webbrowser/Installing')
        self.photo = PhotoImage(file = './Scr/black.png')
        self.style = Style("cyborg").master
        label_l = Label(self,text='Black-Webbrowser',background='gray10',foreground='green',font=("None",30))
        label_l.place(bordermode=INSIDE,x=90,y=20)
        self.install_b = Button(self,text='Install',width=9,command=self.install)
        self.install_b.place(bordermode=OUTSIDE,x=200,y=110)
        self.exit_b = Button(self,text='Exit',width=9,command=self.ext)
        self.exit_b.place(bordermode=OUTSIDE,x=200,y=145)

        self.config(background='white')
        self.iconphoto(False,self.photo)
        self.resizable(False,False)
        self.config(background='gray10')
        self.geometry("500x400")
        self.mainloop()
    def install(self):
        global pr
        pr = Progressbar(self,orient='horizontal',mode='determinate',length=200)
        pr.place(bordermode=INSIDE,x=105,y=65)
        pr.start(55)
        pr.after(6000,self.install_2)
    def install_2(self):
        subprocess.getoutput("chmod a+x black.py")
        pr.stop()
        pr.destroy()
        label_mess = Label(self,text='Complete!',foreground='black',background='white')
        label_mess.place(bordermode=INSIDE,x=170,y=65)
        self.install_b.destroy()
        self.exit_b.place(bordermode=OUTSIDE,x=160,y=110)

    def ext(self):
        self.destroy()
        self.quit()
        quit()
if __name__ == '__main__':
    window = black_webbrowser()


Uninstall/uninstall.py
#!/usr/bin/python3
# black-Webbrowser v1.5
import os,subprocess,platform
try:
    from tkinter import *
    from tkinter.ttk import *
    from tkinter.messagebox import showerror
    from ttkbootstrap import Style
except ImportError:
    subprocess.getoutput("pip install tk-tools")

class black_webbrowser(Tk):
    def __init__(self):
        super(black_webbrowser,self).__init__()
        self.title('Hacker-Webbrowser/Uninstalling')
        self.style = Style("cyborg")
        label_l = Label(self,text='Hacker-Webbrowser',background='white',foreground='black',font=("None",15))
        label_l.place(bordermode=INSIDE,x=120,y=20)
        self.uninstall_b = Button(self,text='uninstall',command=self.uninstall)
        self.uninstall_b.place(bordermode=OUTSIDE,x=160,y=110)
        self.exit_b = Button(self,text='Exit',command=self.ext)
        self.exit_b.place(bordermode=OUTSIDE,x=160,y=135)
        self.photo = PhotoImage(file = './Scr/black/png')
        self.iconphoto(False,self.photo)
        self.config(background='white')
        self.resizable(False,False)
        self.geometry("400x300")
        self.mainloop()
    def uninstall(self):
        global pr
        pr = Progressbar(self,orient='horizontal',mode='determinate',length=200)
        pr.place(bordermode=INSIDE,x=105,y=65)
        pr.start(55)
        pr.after(6000,self.uninstall_2)
    def uninstall_2(self):
        subprocess.getoutput("cd .. && rm -r black-Webbrowser")
        pr.stop()
        pr.destroy()
        label_mess = Label(self,text='Complete!',foreground='black',background='white')
        label_mess.place(bordermode=INSIDE,x=170,y=65)
        self.uninstall_b.destroy()
        self.exit_b.place(bordermode=OUTSIDE,x=160,y=110)

    def ext(self):
        self.destroy()
        self.quit()
        quit()
if __name__ == '__main__':
    if platform.system() == 'Linux':
        if os.getuid() == 0:
          window = black_webbrowser()
        else:
            showerror(title='Cannot Running',message='Please, Check Root!')
    else:
        window = black_webbrowser()
